src/gui/owncloudgui.cpp

```diff
@@ -1063,35 +1063,21 @@ void ownCloudGui::slotShowShareDialog(const QString &sharePath, const QString &l
             maxSharingPermissions = SharePermission(0);
         }
 
-
-        ShareDialog *w = nullptr;
-        if (_shareDialogs.contains(localPath) && _shareDialogs[localPath]) {
-            qCInfo(lcApplication) << "Raising share dialog" << sharePath << localPath;
-            w = _shareDialogs[localPath];
+        if (_shareDialog && _shareDialog->localPath() == localPath) {
+            qCInfo(lcApplication) << "Raising existing share dialog" << sharePath << localPath;
         } else {
-            qCInfo(lcApplication) << "Opening share dialog" << sharePath << localPath << maxSharingPermissions;
-            w = new ShareDialog(accountState, folder->webDavUrl(), sharePath, localPath, maxSharingPermissions, startPage, settingsDialog());
-            w->setAttribute(Qt::WA_DeleteOnClose, true);
-
-            _shareDialogs[localPath] = w;
-            connect(w, &QObject::destroyed, this, &ownCloudGui::slotRemoveDestroyedShareDialogs);
+            qCInfo(lcApplication) << "Opening new share dialog" << sharePath << localPath << maxSharingPermissions;
+            if (_shareDialog) {
+                _shareDialog->close();
```

Comment:
It should also work if we stack them, so should add another one?

Reply:
No, just 1 sharing dialog for the latest. If a user requested one before but never used it to copy the link, then 🤷, apparently it wasn't important.